### PR TITLE
[Component with its own class name as service type] LPS-200037 Refactor Builder and helper class form asset module

### PR DESCRIPTION
**Pull request by MrJohn1911**
Ticket: https://liferay.atlassian.net/browse/LPS-200037

### CODE REVIEW

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:auto-tagging

**Comment by catarineaguiar** [comment]
@MrJohn1911 possible regression:

```
[beanshell]      [exec] com.liferay.document.library.search.test.DLFolderSearchTest > testBaseModelUserPermissions FAILED
[beanshell]      [exec]     java.lang.AssertionError: {"from":0,"size":10000,"query":{"bool":{"must":[{"bool":{"must":[{"bool":{"should":[{"bool":{"must":[{"match":{"comments":{"query":"Title"}}}],"should":[{"match_phrase":{"comments":{"query":"Title","slop":50}}},{"match_phrase":{"comments":{"query":"Title","boost":2.0}}}]}},{"bool":{"must":[{"match":{"content":{"query":"Title"}}}],"should":[{"match_phrase":{"content":{"query":"Title","slop":50}}},{"match_phrase":{"content":{"query":"Title","boost":2.0}}}]}},{"bool":{"must":[{"match":{"description":{"query":"Title"}}}],"should":[{"match_phrase":{"description":{"query":"Title","slop":50}}},{"match_phrase":{"description":{"query":"Title","boost":2.0}}}]}},{"bool":{"must":[{"bool":{"should":[{"match":{"title":{"query":"Title"}}},{"match_phrase_prefix":{"title":{"query":"Title"}}}]}}],"should":[{"match_phrase":{"title":{"query":"Title","boost":2.0}}}]}},{"bool":{"must":[{"match":{"url":{"query":"Title"}}}],"should":[{"match_phrase":{"url":{"query":"Title","slop":50}}},{"match_phrase":{"url":{"query":"Title","boost":2.0}}}]}},{"bool":{"should":[{"match_phrase_prefix":{"userName.text":{"query":"Title"}}}]}},{"bool":{"must":[{"match":{"assetCategoryTitles_en_US":{"query":"Title"}}}],"should":[{"match_phrase":{"assetCategoryTitles_en_US":{"query":"Title","slop":50}}},{"match_phrase":{"assetCategoryTitles_en_US":{"query":"Title","boost":2.0}}}]}},{"bool":{"must":[{"bool":{"should":[{"match":{"assetTagNames_en_US":{"query":"Title"}}},{"match_phrase_prefix":{"assetTagNames_en_US":{"query":"Title"}}}]}}],"should":[{"match_phrase":{"assetTagNames_en_US":{"query":"Title","boost":2.0}}}]}}]}}]}}],"filter":[{"bool":{"must":[{"bool":{"must":[{"terms":{"groupId":["34596"]}},{"terms":{"scopeGroupId":["34596"]}}]}},{"bool":{"should":[{"bool":{"must":[{"term":{"entryClassName":{"value":"com.liferay.document.library.kernel.model.DLFolder"}}},{"term":{"hidden":{"value":"false"}}},{"term":{"hidden":{"value":"false"}}},{"terms":{"status":["0"]}},{"terms":{"status":["0"]}},{"bool":{"should":[{"term":{"userId":{"value":"34604"}}},{"terms":{"roleId":["20101","20105"]}},{"terms":{"sharedToUserId":["34604"]}}]}},{"bool":{"must":[{"bool":{"must_not":[{"exists":{"field":"ctCollectionId"}}]}}]}},{"bool":{"must":[{"bool":{"must_not":[{"exists":{"field":"ctCollectionId"}}]}}]}}]}}]}}]}}]}},"explain":false,"stored_fields":["uid","companyId","entryClassPK","entryClassName","description","title"],"track_scores":true,"track_total_hits":2147483647}->[{uid=[com.liferay.document.library.kernel.model.DLFolder_PORTLET_34599], entryClassPK=[34599], companyId=[4336660651970596073], entryClassName=[com.liferay.document.library.kernel.model.DLFolder], description=[2tYvFjdF], title=[tiTle3vQnJq4hG9BcSDUdU9Gl9tHZsiz2PJKNqwVuxHIX8YrvDPuBGGi1BHV2ft6rQMtJATGalsORSdc6YJWeu8eCxlUxDqqVfnB]}}, length=1, query=null] expected:<0> but was:<1>
[beanshell]      [exec]     	at org.junit.Assert.fail(Assert.java:89)
[beanshell]      [exec]     	at org.junit.Assert.failNotEquals(Assert.java:835)
[beanshell]      [exec]     	at org.junit.Assert.assertEquals(Assert.java:647)
[beanshell]      [exec]     	at com.liferay.portal.search.test.util.BaseSearchTestCase.assertBaseModelsCount(BaseSearchTestCase.java:257)
[beanshell]      [exec]     	at com.liferay.portal.search.test.util.BaseSearchTestCase.testUserPermissions(BaseSearchTestCase.java:1044)
[beanshell]      [exec]     	at com.liferay.portal.search.test.util.BaseSearchTestCase.testBaseModelUserPermissions(BaseSearchTestCase.java:63)
```
full log [here](https://testray.liferay.com/reports/production/logs/2023-11/test-1-37/test-portal-acceptance-pullrequest(master)/15485/modules-integration-mysql57-jdk8/0/9/jenkins-console.txt.gz?authuser=0)

**Comment by MrJohn1911** [comment]
Hey @catarineaguiar I ran this test locally and it did not failed. In fact, this tests belongs to a module that I did not touch in this pull, so I will rerun the relevant just to make sure.

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by zhangyan0701** [comment]
No regressions found. The test passed in the new relevant.

File: modules/apps/asset/asset-publisher-web/src/main/java/com/liferay/asset/publisher/web/internal/portlet/AssetPublisherPortlet.java [hunk 1]
```diff
@@ -493,6 +729,42 @@ private void _getRSS(
 		}
 	}
 
+	private byte[] _getRSSBytes(
```
Comment (rafaprax) [inline modules/apps/asset/asset-publisher-web/src/main/java/com/liferay/asset/publisher/web/internal/portlet/AssetPublisherPortlet.java:732]: Hey @MrJohn1911, please inline the `_getRSSBytes` method. Thx

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:auto-tagging

**Comment by zhangyan0701** [comment]
No regressions found.

**Review by rafaprax — CHANGES_REQUESTED** [review]
Hey @MrJohn1911 , you do not need this commit https://github.com/liferay-core-infra/liferay-portal/pull/6162/commits/0c875979e9a6e3e97bb5666c104ea98f4c08da46

**Comment by zhangyan0701** [comment]
ci:test:auto-tagging

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by zhangyan0701** [comment]
ci:test:auto-tagging

**Comment by zhangyan0701** [comment]
No regressions found.

**Review by rafaprax — APPROVED** [review]

**Comment by tinatian** [comment]
@MrJohn1911 Please rebase.

Thanks.
Tina

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:auto-tagging

**Comment by MrJohn1911** [comment]
ci:report:2488110774

**Comment by MrJohn1911** [comment]
ci:report:2488141592

**Comment by MrJohn1911** [comment]
ci:test:echo

**Comment by MrJohn1911** [comment]
ci:test:echo

**Comment by MrJohn1911** [comment]
ci:report:2488917176

**Comment by MrJohn1911** [comment]
ci:report:2489047049

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:report:2493142535

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:report:2494670966

**Pull request closed without merging**